### ISSUE TITLE
sonar-measures-exports fails with unsupported operand type error with SonarQube 10.1 and non Community Edition

### ISSUE THREAD

**Issue by chamacs** [okorach/sonar-tools#925]
Installed package sonar-tools 2.8, installed using Python 3.11.2.  Run against SonarQube Enterprise 10.1.

command ran:
sonar-measures-export --url "$SONAR_URL" --token "$SONAR_TOKEN" \
    --metricKeys "\
        ncloc, \
        violations, \
        new_violations, \
        bugs, \
        reliability_rating, \
        code_smells, \
        sqale_rating, \
        vulnerabilities, \
        security_rating, \
        security_hotspots, \
        security_review_rating" \
    --percentsAsString \
    --datesWithoutTime \
    --file $OUTPUT_FILE

Error received:
09:11:35 2024-01-03 15:11:35,969 | sonar-tools | INFO    | MainThread      | Computing LoCs
09:11:35 Traceback (most recent call last):
09:11:35   File "/home/jenkins/.local/bin/sonar-measures-export", line 8, in <module>
09:11:35     sys.exit(main())
09:11:35              ^^^^^^
09:11:35   File "/opt/pipx/venvs/sonar-tools/lib/python3.11/site-packages/tools/measures_export.py", line 256, in main
09:11:35     nb_loc += project.loc()
09:11:35               ^^^^^^^^^^^^^
09:11:35   File "/opt/pipx/venvs/sonar-tools/lib/python3.11/site-packages/sonar/projects/projects.py", line 240, in loc
09:11:35     self._ncloc_with_branches = max([b.loc() for b in self.branches().values() + self.pull_requests().values()])
09:11:35                                                       ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
09:11:35 TypeError: unsupported operand type(s) for +: 'dict_values' and 'dict_values'
09:11:36 Build step 'Execute shell' marked build as failure

**Comment by chamacs** [comment]
Last ran successfully with sonar-tools version 2.7.1 against SonarQube Enterprise 9.9.

**Comment by okorach** [comment]
Hello @chamacs,
Thanks for reporting this. Given the stack trace I have doubts that the problem happened with sonar-tools 2.8.
And I see that you use `venv` so there may be a problem of virtual environment on your side.

In order to make sure, can you check the beginning of the logs before the problem happens. It should display which version of sonar-tools is run, something like:
```
sonar-measures-export --metricKeys "ncloc,violations,new_violations,bugs,reliability_rating,code_smells,sqale_rating,vulnerabilities,security_rating,security_hotspots,security_review_rating" --percentsAsString --datesWithoutTime
2024-01-07 14:56:03,809 | sonar-tools | INFO    | MainThread      | Set debug level to INFO
2024-01-07 14:56:03,809 | sonar-tools | INFO    | MainThread      | sonar-tools version 2.8
```

If that's 2.8, can you attach the full logs in DEBUG mode for me to investigate (use option `-v DEBUG`)
If that's an older version, then please make sure you install the latest version 2.8 in your virtual env (note that a 2.8.1 should be released soon to fix a certain number of **other** SonarQube 10.x compatibility problems)

Last but not least, not that `sonar-tools` honor the SonarQube standard `SONAR_TOKEN` and `SONAR_HOST_URL` environment variables, so if they are set, you don't need to pass them on the command line. That would make your command line simpler and shorter.

**Comment by okorach** [comment]
MMhh, my bad, this is indeed **sonar-tools 2.8** and I think I see what the problem is.
It could well be due to Python 3.11 being more strict. Did you also upgrade **Python** when you upgraded **sonar-tools** ?

**Comment by okorach** [comment]
Fixed in sonar-tools 2.8.1 just released

**Comment by chamacs** [comment]
I can confirm that it is now fixed with version 2.8.1.

11:36:23 2024-01-09 17:36:23,467 | sonar-tools | INFO    | MainThread      | Set debug level to INFO
11:36:23 2024-01-09 17:36:23,467 | sonar-tools | INFO    | MainThread      | sonar-tools version 2.8.1
...
11:36:26 2024-01-09 17:36:26,861 | sonar-tools | INFO    | MainThread      | Computing LoCs
11:36:33 2024-01-09 17:36:33,091 | sonar-tools | INFO    | MainThread      | ... PROJECTS ... branches ... LoCs

**Comment by chamacs** [comment]
Thanks for the quick turnaround.
